testmenu.c

```diff
@@ -69,47 +84,83 @@ int main(int argc, char **argv) {
     /* Execute SQL statement */
     rc = sqlite3_exec(db, sql, callback, 0, &zErrMsg);
 
-    if( rc != SQLITE_OK ) {
+    if( rc != SQLITE_OK )
+    {
         fprintf(stderr, "SQL error: %s\n", zErrMsg);
         sqlite3_free(zErrMsg);
     }
 
-    // Print the table
-    sql = "SELECT * FROM USERS";
+    // Print the table users for debug only
+    // sql = "SELECT * FROM USERS";
 
-    rc = sqlite3_exec(db, sql, callback, 0, &zErrMsg);
+    // rc = sqlite3_exec(db, sql, callback, 0, &zErrMsg);
 
-    if( rc != SQLITE_OK ) {
-        fprintf(stderr, "SQL error: %s\n", zErrMsg);
-        sqlite3_free(zErrMsg);
-    }
+    // if( rc != SQLITE_OK )
+    // {
+    //     fprintf(stderr, "SQL error: %s\n", zErrMsg);
+    //     sqlite3_free(zErrMsg);
+    // }
 
-    do {
+    do 
+    {
         // Case 1: User is not connected
-        if(connected == 0) {
+        if(connected == 0)
+        {
             printf("Welcome to your glycemia database, please choose an option:\n");
             printf("1. Log in with your user\n");
             printf("2. Create a new user\n");
-            printf("3. Exit\n");
+            printf("9. Exit\n");
             scanf("%d", &choice);
 
-            if(choice == 1) {
-                printf("Please enter your username:\n");
+            if(choice == 1)
+            {
+                printf("\nPlease enter your username:\n");
                 scanf("%s", &username);
+                strcat(username, "\0");
+                printf("\n");
+
                 printf("Please enter your password:\n");
-                scanf("%s", &password);
-                isUserValid = LogIn(username, password, db); 
-                if(isUserValid) {
-                    printf("You are now connected!\n");
+                fflush(stdin);
+                char* input = malloc(30);
+                char* getPassword = malloc(30);
+                getPassword = fgets(input, 30, stdin);
+                for(int i = 0; i < strlen(getPassword); i++)
+                {
+                    if(getPassword[i] == '\n')
+                    {
+                        getPassword[i] = getPassword [i + 1];
+                    }
+                }
+                printf("\n");
+
+                char sql[200] = "SELECT * FROM USERS WHERE username = '";
+                strcat(sql, username);
+                strcat(sql, "' AND password = '");
+                strcat(sql, getPassword);
+                strcat(sql, "'");
+
+                rc = sqlite3_exec(db, sql, checkUser, 0, &zErrMsg);
+
+                if( rc == SQLITE_OK )
+                {
+                    printf("Wrong username or password, please try again\n\n");
+                    sqlite3_free(zErrMsg);
```

Comment:
free(): double free detected in tcache 2
Aborted (core dumped) 
-> when i put a random password with a random (not created user) i get this error.  
->Before it told me WRong password or username so it's working but smth is freed twice

Reply:
we will see together in can't repro the error